drivers/shared/executor/executor.go

```diff
@@ -275,17 +233,16 @@ func (e *UniversalExecutor) Launch(command *ExecCommand) (*ProcessState, error)
 		return nil, err
 	}
 
-	stdout, err := e.commandCfg.Stdout()
-	if err != nil {
-		return nil, err
-	}
-	stderr, err := e.commandCfg.Stderr()
+	pio, err := procio.NewProcessIO(e.commandCfg.stdio(),
+		func(out, err io.WriteCloser) {
+			e.childCmd.Stdout = out
+			e.childCmd.Stderr = err
+		})
```

Comment:
I'd suggest using the getters instead of adding callbacks and a callback type.  The following seems just as expressive without callback overhead:

```go
pio, err := procio.NewProcessIO(e.commandCfg.stdio())
if err != nil { return nil, err }
e.childCmd.Stdout, e.childCmd.Stderr = pio.IO().Stdout(), pio.IO().Stderr()
```

Reply:
Also, setting `e.childCmd.Std{out,err}` to the correct type (`*os.File`) is the critical part of this diff for getting the proper `childCmd.Wait` behavior, right? If so let's call that out with a big huge comment.